code/lib/cli-storybook/src/automigrate/fixes/consolidated-imports.ts

```diff
@@ -0,0 +1,199 @@
+
+import pLimit from 'p-limit';
+import prompts from 'prompts';
+import { dedent } from 'ts-dedent';
+
+import { consolidatedPackages } from '../helpers/consolidated-packages';
+import type { Fix, RunOptions } from '../types';
+
+export interface ConsolidatedOptions {
+  packageJsonFiles: string[];
+}
+
+function transformPackageJson(content: string): string | null {
+  const packageJson = JSON.parse(content);
+  let hasChanges = false;
+
+  // Check dependencies
+  if (packageJson.dependencies) {
+    for (const [dep, version] of Object.entries(packageJson.dependencies)) {
+      if (dep in consolidatedPackages) {
+        delete packageJson.dependencies[dep];
+        hasChanges = true;
+      }
+    }
+  }
+
+  // Check devDependencies
+  if (packageJson.devDependencies) {
+    for (const [dep, version] of Object.entries(packageJson.devDependencies)) {
+      if (dep in consolidatedPackages) {
+        delete packageJson.devDependencies[dep];
+        hasChanges = true;
+      }
+    }
+  }
+
+  return hasChanges ? JSON.stringify(packageJson, null, 2) : null;
+}
+
+function transformImports(source: string) {
+  let hasChanges = false;
+  let transformed = source;
+
+  for (const [from, to] of Object.entries(consolidatedPackages)) {
+    // Match the package name when it's inside either single or double quotes
+    const regex = new RegExp(`(['"])${from}\\1`, 'g');
+    if (regex.test(transformed)) {
+      transformed = transformed.replace(regex, `$1${to}$1`);
+      hasChanges = true;
+    }
+  }
+
+  return hasChanges ? transformed : null;
+}
+
+export const transformPackageJsonFiles = async (files: string[], dryRun: boolean) => {
+  const errors: Array<{ file: string; error: Error }> = [];
+  const limit = pLimit(10);
+
+  await Promise.all(
+    files.map((file) =>
+      limit(async () => {
+        try {
+          const contents = await readFile(file, 'utf-8');
+          const transformed = transformPackageJson(contents);
+          if (!dryRun && transformed) {
+            await writeFile(file, transformed);
+          }
+        } catch (error) {
+          errors.push({ file, error: error as Error });
+        }
+      })
+    )
+  );
+
+  return errors;
+};
+
+export const transformImportFiles = async (files: string[], dryRun: boolean) => {
+  const errors: Array<{ file: string; error: Error }> = [];
+  const limit = pLimit(10);
+
+  await Promise.all(
+    files.map((file) =>
+      limit(async () => {
+        try {
+          const contents = await readFile(file, 'utf-8');
+          const transformed = transformImports(contents);
+          if (!dryRun && transformed) {
+            await writeFile(file, transformed);
+          }
+        } catch (error) {
+          errors.push({ file, error: error as Error });
+        }
+      })
+    )
+  );
+
+  return errors;
+};
+
+export const consolidatedImports: Fix<ConsolidatedOptions> = {
+  id: 'consolidated-imports',
+  versionRange: ['^8.0.0', '^9.0.0'],
```

Comment:
```suggestion
  versionRange: ['^8.0.0', '^9.0.0-0'],
```

To support beta releases, otherwise we will have a hard time to test this.